src/test/java/net/imglib2/algorithm/hough/HoughLineTransformTest.java

```diff
@@ -0,0 +1,185 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imglib2.algorithm.hough;
+
+import static org.junit.Assert.assertEquals;
+
+import java.awt.image.BufferedImage;
+import java.awt.image.DataBufferByte;
+import java.io.IOException;
+import java.util.List;
+
+import javax.imageio.ImageIO;
+
+import net.imglib2.Point;
+import net.imglib2.RandomAccess;
+import net.imglib2.img.Img;
+import net.imglib2.img.array.ArrayImgFactory;
+import net.imglib2.img.array.ArrayImgs;
+import net.imglib2.type.numeric.RealType;
+import net.imglib2.type.numeric.integer.UnsignedByteType;
+import net.imglib2.util.Util;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests {@link HoughTransforms#voteLines}, {@link HoughTransforms#pickPeaks},
+ * {@link HoughTransforms#getSlope}, and {@link HoughTransforms#getIntercept}.
+ *
+ * @author Yili Zhao
+ * @author Gabe Selzer
+ */
+public class HoughLineTransformTest
+{
+
+	private BufferedImage groundTruth;
+
+	private Img< UnsignedByteType > tpImg;
+
+	public static int N_THREADS = Runtime.getRuntime().availableProcessors();
+
+	@Before
+	public void setUp() throws IOException
+	{
+		groundTruth = ImageIO.read( getClass().getResource( "result9.png" ) );
+		tpImg = ArrayImgs.unsignedBytes( 11l, 13l );
+		final RandomAccess< UnsignedByteType > randomAccess = tpImg.randomAccess();
+		// create vertical line
+		randomAccess.setPosition( 5, 0 );
+		for ( int i = 0; i < 13; i++ )
+		{
+			randomAccess.setPosition( i, 1 );
+			randomAccess.get().set( 255 );
+		}
+
+		// create horizontal line
+		randomAccess.setPosition( 5, 1 );
+		for ( int i = 0; i < 11; i++ )
+		{
+			randomAccess.setPosition( i, 0 );
+			randomAccess.get().set( 255 );
+		}
+
+		// create diagonal line
+		for ( int i = 0; i < 11; i++ )
+		{
+			randomAccess.setPosition( i, 0 );
+			randomAccess.setPosition( i / 3, 1 );
+			randomAccess.get().set( 255 );
+		}
+	}
+
+	@Test
+	public < T extends RealType< T > > void testHoughLineTransformToTarget() throws IOException
+	{
+		// create votespace
+		final long[] dims = new long[ tpImg.numDimensions() ];
+		tpImg.dimensions( dims );
+		final long[] outputDims = HoughTransforms.getVotespaceSize( tpImg );
+		final Img< UnsignedByteType > votespace = new ArrayImgFactory().create( outputDims, tpImg.firstElement() );
```

Comment:
Create `byte[] store` of appropriate size and replace `new ArrayImgFactory().create` by `ArrayImgs.unsignedBytes( store, outputDims )`. Then, you can use `Assert.assertArrayEquals( expected, store )` instead of iterating over pixels.